doc/faq.rst

```diff
@@ -378,6 +378,100 @@ password fields give no feedback, not even whether a character has been typed.
+       Cloning into 'charliecloud'...
+       Warning: Permanently added 'github.com' (RSA) to the list of known hosts.
+       remote: Enumerating objects: 10337, done.
+       remote: Counting objects: 100% (601/601), done.
+       remote: Compressing objects: 100% (299/299), done.
+       remote: Total 10337 (delta 400), reused 441 (delta 297), pack-reused 9736
+       Receiving objects: 100% (10337/10337), 12.77 MiB | 22.31 MiB/s, done.
+       Resolving deltas: 100% (7061/7061), done.
+       grown in 4 instructions: foo
+
+     Check your institutional policy on whether this is permissible, though
+     it's worth noting that users `almost never
+     <https://www.usenix.org/system/files/login/articles/105484-Gutmann.pdf>`_
+     verify the host fingerprints anyway.
+
+     This will not record details of the connection in the image.
+
+     More complex edit commands will be needed if either of these directives
+     already appears in :code:`/etc/ssh/ssh_config`, because SSH config uses
+     the first instance of a directive it encounters.
+
+  3. Add the remote host to the system known hosts file, e.g.::
+
+       $ cat ./Dockerfile
+       FROM alpine:latest
+       RUN apk add git openssh
+       RUN echo 'github.com,140.82.112.4 ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==' >> /etc/ssh/ssh_known_hosts
+       RUN git clone git@github.com:hpc/charliecloud.git
+       $ ch-image build -t foo -f ./Dockerfile .
+       [...]
+         4 RUN ['/bin/sh', '-c', 'git clone git@github.com:hpc/charliecloud.git']
+       Cloning into 'charliecloud'...
+       remote: Enumerating objects: 10337, done.
+       remote: Counting objects: 100% (601/601), done.
+       remote: Compressing objects: 100% (299/299), done.
+       remote: Total 10337 (delta 400), reused 441 (delta 297), pack-reused 9736
+       Receiving objects: 100% (10337/10337), 12.77 MiB | 24.57 MiB/s, done.
+       Resolving deltas: 100% (7061/7061), done.
+       grown in 4 instructions: foo
+
+     This records connection details in both the Dockerfile and the image.
+
+.. note:: This FAQ does not cover the related issue of SSH authentication. One
+          approach is to simply run the SSH agent on the host.
+          :code:`ch-image` passes environment variable :code:`SSH_AUTH_SOCK`
+          into the build and bind-mounts host :code:`/tmp` to guest
+          :code:`/tmp`, which is where the SSH agent's listening socket
+          usually resides. Thus, SSH within the container will use this
+          existing SSH agent on the host to authenticate without further
+          intervention.
+
 
```

Comment:
I'm not sure how clever/crazy you want this to get, but some other possibilities you may want to consider at least mentioning:
 - Git credentials (see [gitcredentials(7)](https://git-scm.com/docs/gitcredentials) and related subcommand/helper docs)
 - [Personal Access Tokens](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token) or equivalent
 - `SSH_ASKPASS`, `GIT_ASKPASS`, the GnuPG agent's `pinentry-program`, et al., for out-of-band password/PIN interrogation/caching options
 - Automatic host key fingerprint verification - [SSHFP via DNSSEC](https://weberblog.net/sshfp-authenticate-ssh-fingerprints-via-dnssec/)

There are potentially other options as well (process substitution, socket multiplexing, etc.), depending on how much one might trust the container image, but the above at least should be robust enough to qualify as "safe," IMHO, with varying degrees of simplicity/reliability/dad-proofing....

Reply:
Also good points and in issue #1120.